src/Employer/Employer.Web/Orchestrators/Part1/LocationOrchestrator.cs

```diff
@@ -152,41 +179,86 @@ private async Task<LocationViewModel> GetViewModelAsync(Vacancy vacancy, bool? h
-                    vm.Postcode = vacancy.EmployerLocation.Postcode;
+                    var employerLocation = vacancy.EmployerLocation.ToString();
+                    StringComparer comparer = StringComparer.OrdinalIgnoreCase;
+                    var otherLocations = vm.OtherLocationsAddress;
+                    if (otherLocations?.Any() == true)
+                    {
+                        foreach (var location in otherLocations)
+                        {
+                            if (comparer.Compare(employerLocation, location.ToString()) == 0)
+                            {
+                                vm.Location = location.ToString();
+                            }
+                        }
+                    }
                 }
-            }            
+            }
+
             return vm;
-        }        
+        }
 
-       private async Task UpdateEmployerProfileAsync(string employerAccountId, long legalEntityId, string tradingName, VacancyUser user)
+        private async Task UpdateEmployerProfileAsync(string employerAccountId,
+            VacancyEmployerInfoModel employerInfoModel, LocationEditModel locationEditModel, VacancyUser user)
         {
+            var legalEntityId = employerInfoModel.LegalEntityId.GetValueOrDefault();
+            var tradingName = employerInfoModel.NewTradingName;
+
             var employerProfile =
                 await _recruitVacancyClient.GetEmployerProfileAsync(employerAccountId, legalEntityId);
 
             if (employerProfile == null)
             {
-                throw new NullReferenceException($"No Employer Profile was found for employerAccount: {employerAccountId}, legalEntity: {legalEntityId}");
+                throw new NullReferenceException(
+                    $"No Employer Profile was found for employerAccount: {employerAccountId}, legalEntity: {legalEntityId}");
             }
 
-            if (employerProfile.TradingName != tradingName)
+            if (employerInfoModel.EmployerNameOption == EmployerNameOptionViewModel.NewTradingName &&
+                employerProfile.TradingName != tradingName)
+                employerProfile.TradingName = tradingName;               
+
+        if (locationEditModel.Location == UseOtherLocation)
+        {
+            var address = new Address {
+                AddressLine1 = locationEditModel.AddressLine1,
+                AddressLine2 = locationEditModel.AddressLine2,
+                AddressLine3 = locationEditModel.AddressLine3,
+                AddressLine4 = locationEditModel.AddressLine4,
+                Postcode = locationEditModel.Postcode
+            };     
+            if(CheckForDuplicates(employerProfile.OtherLocations,address))
             {
-                employerProfile.TradingName = tradingName;
-                await _recruitVacancyClient.UpdateEmployerProfileAsync(employerProfile, user);
-            }
+                employerProfile.OtherLocations.Add(address);
```

Comment:
`CheckForDuplicates` suggests that employerProfile.OtherLocations could be null. If this is the case then this will break when we try to `Add`.